### ISSUE TITLE
Changing the name of operator will result in different performance ?

### ISSUE THREAD

**Issue by gclouding** [apache/incubator-mxnet#2383]
First, I trained the mnist using mlp which is provided in the example.

def get_mlp():
    """
    multi-layer perceptron
    """
    data = mx.symbol.Variable('data')
    fc1  = mx.symbol.FullyConnected(data = data, name='fc1', num_hidden=128)
    act1 = mx.symbol.Activation(data = fc1, name='relu1', act_type="relu")
    fc2  = mx.symbol.FullyConnected(data = act1, name = 'fc2', num_hidden = 64)
    act2 = mx.symbol.Activation(data = fc2, name='relu2', act_type="relu")
    #xx = mx.symbol.Activation(data = act2, name='xx', act_type = "relu")
    fc3  = mx.symbol.FullyConnected(data = act2, name='fc3', num_hidden=10)
    mlp  = mx.symbol.SoftmaxOutput(data = fc3, name = 'softmax')
    return mlp

And the training result (batch size =5000)is:
2016-06-09 16:15:12,234 Node[0] Epoch[0] Batch [2]  Speed: 598587.70 samples/sec    Train-accuracy=0.078700
2016-06-09 16:15:12,234 Node[0] Epoch[0] Batch [2]  Speed: 598587.70 samples/sec    Train-top_k_accuracy_5=0.407000
2016-06-09 16:15:12,243 Node[0] Epoch[0] Batch [3]  Speed: 564387.75 samples/sec    Train-accuracy=0.093933
2016-06-09 16:15:12,243 Node[0] Epoch[0] Batch [3]  Speed: 564387.75 samples/sec    Train-top_k_accuracy_5=0.450733
2016-06-09 16:15:12,252 Node[0] Epoch[0] Batch [4]  Speed: 610666.82 samples/sec    Train-accuracy=0.135350
2016-06-09 16:15:12,252 Node[0] Epoch[0] Batch [4]  Speed: 610666.82 samples/sec    Train-top_k_accuracy_5=0.520150
2016-06-09 16:15:12,260 Node[0] Epoch[0] Batch [5]  Speed: 594464.54 samples/sec    Train-accuracy=0.188640
2016-06-09 16:15:12,261 Node[0] Epoch[0] Batch [5]  Speed: 594464.54 samples/sec    Train-top_k_accuracy_5=0.584440
2016-06-09 16:15:12,269 Node[0] Epoch[0] Batch [6]  Speed: 574326.17 samples/sec    Train-accuracy=0.240900
2016-06-09 16:15:12,269 Node[0] Epoch[0] Batch [6]  Speed: 574326.17 samples/sec    Train-top_k_accuracy_5=0.639267
2016-06-09 16:15:12,278 Node[0] Epoch[0] Batch [7]  Speed: 577966.65 samples/sec    Train-accuracy=0.285257
2016-06-09 16:15:12,278 Node[0] Epoch[0] Batch [7]  Speed: 577966.65 samples/sec    Train-top_k_accuracy_5=0.682543
2016-06-09 16:15:12,287 Node[0] Epoch[0] Batch [8]  Speed: 615235.13 samples/sec    Train-accuracy=0.317750
2016-06-09 16:15:12,287 Node[0] Epoch[0] Batch [8]  Speed: 615235.13 samples/sec    Train-top_k_accuracy_5=0.716100
2016-06-09 16:15:12,295 Node[0] Epoch[0] Batch [9]  Speed: 593052.43 samples/sec    Train-accuracy=0.342356
2016-06-09 16:15:12,295 Node[0] Epoch[0] Batch [9]  Speed: 593052.43 samples/sec    Train-top_k_accuracy_5=0.742756
2016-06-09 16:15:12,304 Node[0] Epoch[0] Batch [10] Speed: 595731.05 samples/sec    Train-accuracy=0.363100
2016-06-09 16:15:12,304 Node[0] Epoch[0] Batch [10] Speed: 595731.05 samples/sec    Train-top_k_accuracy_5=0.764460
2016-06-09 16:15:12,313 Node[0] Epoch[0] Batch [11] Speed: 555007.67 samples/sec    Train-accuracy=0.385164
2016-06-09 16:15:12,313 Node[0] Epoch[0] Batch [11] Speed: 555007.67 samples/sec    Train-top_k_accuracy_5=0.782618
2016-06-09 16:15:12,321 Node[0] Epoch[0] Batch [12] Speed: 617972.65 samples/sec    Train-accuracy=0.408717
2016-06-09 16:15:12,322 Node[0] Epoch[0] Batch [12] Speed: 617972.65 samples/sec    Train-top_k_accuracy_5=0.798267

Then, I just change the name of each operator defined in the mlp network as follows:
def get_mlp():
    """
    multi-layer perceptron
    """
    data = mx.symbol.Variable('data')
    fc1  = mx.symbol.FullyConnected(data = data, name='fc0', num_hidden=128)
    act1 = mx.symbol.Activation(data = fc1, name='relu1', act_type="relu")
    fc2  = mx.symbol.FullyConnected(data = act1, name = 'fc1', num_hidden = 64)
    act2 = mx.symbol.Activation(data = fc2, name='relu2', act_type="relu")
    fc3  = mx.symbol.FullyConnected(data = act2, name='fc2', num_hidden=10)
    mlp  = mx.symbol.SoftmaxOutput(data = fc3, name = 'softmax')
    return mlp

But, the result is different in terms of accuracy:
2016-06-09 16:15:50,184 Node[0] Epoch[0] Batch [2]  Speed: 507048.36 samples/sec    Train-accuracy=0.133200
2016-06-09 16:15:50,184 Node[0] Epoch[0] Batch [2]  Speed: 507048.36 samples/sec    Train-top_k_accuracy_5=0.563300
2016-06-09 16:15:50,193 Node[0] Epoch[0] Batch [3]  Speed: 554318.19 samples/sec    Train-accuracy=0.164000
2016-06-09 16:15:50,193 Node[0] Epoch[0] Batch [3]  Speed: 554318.19 samples/sec    Train-top_k_accuracy_5=0.603600
2016-06-09 16:15:50,202 Node[0] Epoch[0] Batch [4]  Speed: 551084.48 samples/sec    Train-accuracy=0.205200
2016-06-09 16:15:50,203 Node[0] Epoch[0] Batch [4]  Speed: 551084.48 samples/sec    Train-top_k_accuracy_5=0.648200
2016-06-09 16:15:50,211 Node[0] Epoch[0] Batch [5]  Speed: 588178.94 samples/sec    Train-accuracy=0.238960
2016-06-09 16:15:50,211 Node[0] Epoch[0] Batch [5]  Speed: 588178.94 samples/sec    Train-top_k_accuracy_5=0.690360
2016-06-09 16:15:50,221 Node[0] Epoch[0] Batch [6]  Speed: 532258.57 samples/sec    Train-accuracy=0.264767
2016-06-09 16:15:50,221 Node[0] Epoch[0] Batch [6]  Speed: 532258.57 samples/sec    Train-top_k_accuracy_5=0.727133
2016-06-09 16:15:50,230 Node[0] Epoch[0] Batch [7]  Speed: 584930.69 samples/sec    Train-accuracy=0.291400
2016-06-09 16:15:50,230 Node[0] Epoch[0] Batch [7]  Speed: 584930.69 samples/sec    Train-top_k_accuracy_5=0.755857
2016-06-09 16:15:50,239 Node[0] Epoch[0] Batch [8]  Speed: 559852.64 samples/sec    Train-accuracy=0.322050
2016-06-09 16:15:50,239 Node[0] Epoch[0] Batch [8]  Speed: 559852.64 samples/sec    Train-top_k_accuracy_5=0.779550
2016-06-09 16:15:50,248 Node[0] Epoch[0] Batch [9]  Speed: 551084.48 samples/sec    Train-accuracy=0.353889
2016-06-09 16:15:50,248 Node[0] Epoch[0] Batch [9]  Speed: 551084.48 samples/sec    Train-top_k_accuracy_5=0.799289
2016-06-09 16:15:50,257 Node[0] Epoch[0] Batch [10] Speed: 561802.35 samples/sec    Train-accuracy=0.384780
2016-06-09 16:15:50,257 Node[0] Epoch[0] Batch [10] Speed: 561802.35 samples/sec    Train-top_k_accuracy_5=0.815900
2016-06-09 16:15:50,266 Node[0] Epoch[0] Batch [11] Speed: 552129.11 samples/sec    Train-accuracy=0.412218
2016-06-09 16:15:50,266 Node[0] Epoch[0] Batch [11] Speed: 552129.11 samples/sec    Train-top_k_accuracy_5=0.830073
2016-06-09 16:15:50,276 Node[0] Epoch[0] Batch [12] Speed: 535958.50 samples/sec    Train-accuracy=0.438067
2016-06-09 16:15:50,276 Node[0] Epoch[0] Batch [12] Speed: 535958.50 samples/sec    Train-top_k_accuracy_5=0.842033

I have tried many times and found the difference always exists. So I am wondering why just changing the name of operator will result in the difference in accuracy...??

**Comment by phunterlau** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
